### PR TITLE
Destination BigQuery:  Consolidation of objects to StreamConfig, cleanup

### PR DESCRIPTION
**Pull request by gisripa**
## What
Removing redundant references and duplicate information passed around using `WriteConfig` objects. No functional changes and resurrected all the information needed through `StreamConfig` and adapted changes accordingly. 

This PR should be in a mergeable state with no functional changes after the ones down the stack are published. 

## Review guide
* Removed references of `BigQueryWriteConfig` and reused already built `StreamConfig`
* Removing unnecessary `StagingOperations` interface and made concrete class, this will help for later adding a shim on this and refactoring without large changes 
* Removed other unnecessary references of getting dynamic schema, `WriteDispostion` etc. Probably remnant of bigquery-denormalized bespoke connector. 

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

### CODE REVIEW

**Comment by gisripa** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38131?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#38131** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38131?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#38107** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38107?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#38121** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38121?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#38067](https://github.com/airbytehq/airbyte/pull/38067) <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38067?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* **#38127** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38127?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 1 other dependent PR ([#38126](https://github.com/airbytehq/airbyte/pull/38126) <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/38126?utm_source=stack-comment-icon" target="_blank">Graphite</a>)
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gisripa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java [hunk 1]
```diff
@@ -234,7 +236,8 @@ public SerializedAirbyteMessageConsumer getSerializedMessageConsumer(final JsonN
     final String datasetLocation = BigQueryUtils.getDatasetLocation(config);
     final BigQuerySqlGenerator sqlGenerator = new BigQuerySqlGenerator(config.get(BigQueryConsts.CONFIG_PROJECT_ID).asText(), datasetLocation);
     final Optional<String> rawNamespaceOverride = TypingAndDedupingFlag.getRawNamespaceOverride(RAW_DATA_DATASET);
-    final ParsedCatalog parsedCatalog = parseCatalog(config, catalog, datasetLocation, rawNamespaceOverride);
+    final ParsedCatalog parsedCatalog = parseCatalog(sqlGenerator, defaultNamespace,
+        rawNamespaceOverride.orElse(JavaBaseConstants.DEFAULT_AIRBYTE_INTERNAL_NAMESPACE), catalog);
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java:240]: random thought: now that we're in kotlin, do you want to make catalogparser accept `rawNamespace: String?` and do `rawNamespace ?: DEFAULT_AIRBYTE_INTERNAL`? then we don't need to copy this logic into every connector

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java [hunk 2]
```diff
@@ -79,71 +73,52 @@ private long getBigQueryBufferMemoryLimit() {
     return (long) (Runtime.getRuntime().maxMemory() * 0.4);
   }
 
-  private Map<StreamDescriptor, BigQueryWriteConfig> createWriteConfigs(final JsonNode config,
-                                                                        final ConfiguredAirbyteCatalog catalog,
-                                                                        final ParsedCatalog parsedCatalog,
-                                                                        final Function<JsonNode, BigQueryRecordFormatter> recordFormatterCreator,
-                                                                        final Function<String, String> tmpTableNameTransformer) {
+  private Map<StreamDescriptor, StreamConfig> createWriteConfigs(final ConfiguredAirbyteCatalog catalog,
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java:76]: can this just be `parsedCatalog.getStreams().stream()`? afaict we don't actually need the raw protocol models for anything

(and then we don't need to plumb the raw configured catalog into this method)

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java [hunk 3]
```diff
@@ -79,71 +73,52 @@ private long getBigQueryBufferMemoryLimit() {
...
 
-      for (final BigQueryWriteConfig writeConfig : writeConfigs.values()) {
+      for (final StreamConfig streamConfig : streamConfigs) {
+        final var tableId = TableId.of(streamConfig.getId().getRawNamespace(), streamConfig.getId().getRawName());
         LOGGER.info("Preparing staging are in destination for schema: {}, stream: {}, target table: {}, stage: {}",
-            writeConfig.tableSchema(), writeConfig.streamName(), writeConfig.targetTableId(), writeConfig.streamName());
-        // In Destinations V2, we will always use the 'airbyte' schema/namespace for raw tables
+            BigQueryRecordFormatter.SCHEMA_V2, streamConfig.getId().getOriginalName(),
+            tableId, streamConfig.getId().getOriginalName());
+        // In Destinations V2, we will always use the 'airbyte' schema/originalNamespace for raw tables
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java:107]:
```suggestion
        // In Destinations V2, we will always use the 'airbyte_internal' schema/originalNamespace for raw tables
```
😅

File: airbyte-integrations/connectors/destination-bigquery/src/test/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigqueryDestinationHandlerTest.java [hunk 4]
```diff
@@ -57,11 +56,10 @@ public void testToDialectType() {
   @Test
   public void testClusteringMatches() {
     StreamConfig stream = new StreamConfig(mock(),
-        SyncMode.INCREMENTAL,
         DestinationSyncMode.APPEND_DEDUP,
         List.of(new ColumnId("foo", "bar", "fizz")),
         Optional.empty(),
-        new LinkedHashMap<>());
+        new LinkedHashMap<>(), 0, 0, 0);
```
Comment (edgao) [inline airbyte-integrations/connectors/destination-bigquery/src/test/java/io/airbyte/integrations/destination/bigquery/typing_deduping/BigqueryDestinationHandlerTest.java:62]: .... do you want me to set default values for the generation/sync ID args? These diffs seem kind of dumb

**Review by edgao — APPROVED** [review]
couple minor comments, nothing blocking (and some of them are about the CDK PR this is stacked on :P )

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java (hunk 1 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryDestination.java:240]: Yeah makes sense to me.

File: airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java (hunk 3 above)
Comment (gisripa) [reply airbyte-integrations/connectors/destination-bigquery/src/main/java/io/airbyte/integrations/destination/bigquery/BigQueryStagingConsumerFactory.java]: fixed.

**Pull request merged**